### DISCUSSION TITLE
Icon for home assistant

### DISCUSSION THREAD

**Discussion by bilogic** [arendst/Tasmota#21435] [Q&A]
Is there a way to specify the on/off icons in Tasmota?

This makes it much easier to have icon changes rather than doing it in HA using another entity with several lines of templating

**Comment by sfromis** [comment]
Short answer: Nope, not in current Tasmota. You could config *text* for buttons using the `WebButton` command, but that's not images.

Longer answer: If you build page elements in HTML code, and add such to a page, images are possible. Of course, that not something simple which you just do without diving a bit deeper.

**Reply by bilogic** [reply]
actually HA is not asking for images (jpg/png), more like text in the form of `mdi:power-on` and `mdi:power-off` as part of the MQTT message

**Reply by sfromis** [reply]
It sounded like you wanted to have icons *in Tasmota*.....

But if you meant "the" MQTT message, it would have made sense to describe *which* specific one you're talking about, and what you'd like to have.....

The general answer is that there is no feature to add stuff to *arbitrary* MQTT messages, and anything would depend on the specifics. Of course, if you just want *additional* messages, you can do whatever.

**Comment by joba-1** [comment]
This, slightly modified, will probably help:

https://tasmota.github.io/docs/Rules/#send-mqtt-message-on-button-press

**Reply by bilogic** [reply]
![image](https://github.com/arendst/Tasmota/assets/946010/ed50419e-34ee-4de1-911d-2bd537892c36)

I wrote an MQTT client in PHP based off Tasmota, which is able to send different icons based on its current state, the key of interest is `icon`, and the `config` payload consist of many fields to send.

Is there a way to just send only `icon`? My MQTT knowledge is purely based on modifying what I saw, pretty shallow.

**Reply by sfromis** [reply]
Well, using the suggested technique, you can generate *additional* messages with the payload you choose, details are up to you. But first you have to know what data you'd even expect *in the MQTT messages* which is dissimilar to the tiny screenshot snippet you cut.

**Reply by bilogic** [reply]
Ok, it might have been a mix of tasmota and others and derived by observing the messages sent to the MQTT broker, but pretty sure `/config` came from Tasmota.

My thinking was to see if it is possible to add 2 more text strings per switch, much like how we also name the Tasmota device itself. (Since Tasmota is able to talk to HA using MQTT already)

Let me try it out, but I don't think HA is going to accept a payload with just the icon key.

**Reply by sfromis** [reply]
While Tasmota may be involved, **nothing** in your severely cut down screenshot (copy/paste works much better) is recognizable as mapped to Tasmota data. Not a single string or name is like what Tasmota is publishing. Only vaguely similar thing is the word "config" alone.

Whatever you can/should do with HA is not really the topic here, that's why I underline the need for being sure of exactly what you want to get **in the MQTT messages**. Only after having that, it is feasible to suggest how to do that in Tasmota.

Note: This is not related to having any MQTT knowledge, as this merely acts as a transport protocol. The specific data transferred is up what is done in Tasmota, and up to what is done in HA to be able to receive and parse the data for useful purposes.
